pkg/catalog/catalog.go

```diff
@@ -1951,24 +1943,17 @@ func (c *Catalog) PrepareGCUncommitted(ctx context.Context, repositoryID string,
 	// Write parquet to local storage
 	hasData := false
 	var newMark *GCUncommittedMark
-	if mark != nil && mark.BranchID == "" {
-		newMark, hasData, err = c.writeTokensLocal(ctx, repository, uw, pw, mark, runID)
+	if mark == nil || mark.BranchID != "" {
+		newMark, hasData, err = c.writeUncommittedLocal(ctx, repository, uw, pw, mark, runID)
 		if err != nil {
 			return nil, err
 		}
-	} else {
-		newMark, hasData, err = c.writeUncommittedLocal(ctx, repository, uw, pw, mark, runID)
+	}
+	if newMark == nil {
+		newMark, hasData, err = c.writeTokensLocal(ctx, repository, uw, pw, mark, runID, hasData)
```

Comment:
You are assigning hasData and passing it to the function, you can just pass it by reference and save on the returned values

Reply:
If I'm passing it by reference I need to assign it in addition to the return value. I'm not sure it's better

Reply:
Lets sync